libraries/AP_Airspeed/AP_Airspeed.cpp

```diff
@@ -393,11 +393,31 @@ void AP_Airspeed::read(uint8_t i)
         // use an IIR filter, otherwise a bad reading will last for
         // some time after the sensor becomees healthy again
         state[i].filtered_pressure = airspeed_pressure;
+        state[i].IAS_filtered_pressure = airspeed_pressure;
+        state[i].IAS_integ = 0;
     } else {
         state[i].filtered_pressure = 0.7f * state[i].filtered_pressure + 0.3f * airspeed_pressure;
+
+        // apply a second order complementary filter (from TECS) to airspeed_pressure
```

Comment:
This isn't an exact lift of the TECS filter. You could ignore the EAS2TAS part since that factors out, but you are applying the filter here to the pressure, which is normally square rooted before TECS saw it. This leads to a different response time on the filter.

Reply:
yes, it looks like this was the wrong way to go; 
will have to toss this and do a near verbatim copy of the TECS code to be safe.